### PR TITLE
Enable forwarded for prefix header configuration

### LINKED ISSUE #9622: Support `X-Forwarded-Prefix`
**Description**
My application runs behind a reverse proxy and I need to get the original path from `UriInfo` to return it as a part of the `Location`-header in a REST-resource. There already is support for several `X-Forwarded-`-headers, but not for `X-Forwarded-Prefix`.

It would be nice to have support for this header. Maybe it is even possible to make the header-name configurable? Because `X-Forwarded-Prefix` is not a standard and I'm using envoy, which only provides `x-envoy-original-path`.

I've seen that `ForwardedParser` is inspired by the Spring-solution. Spring seems to have support for `X-Forwarded-Prefix`.

### PR DESCRIPTION
**Pull request by ejba**
This is the last part of #9622 that enable users to configure the header to be used to prefix the HTTP request's URI. It includes also documentation how to setup it.

Fixes: #9622

### CODE REVIEW

File: docs/src/main/asciidoc/vertx.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/vertx.adoc:710]:
```suggestion
In those scenarios, Quarkus can be configured to automatically update information like protocol, host, port and URI
```

File: docs/src/main/asciidoc/vertx.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/vertx.adoc:708]:
```suggestion
Quarkus could be accessed through proxies that additionally generate headers (e.g. `X-Forwarded-Host`) to keep
```

File: docs/src/main/asciidoc/vertx.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/vertx.adoc:713]:
```suggestion
IMPORTANT: Activating this feature makes the server exposed to several security issues (i.e. information spoofing).
```

File: docs/src/main/asciidoc/vertx.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/vertx.adoc:737]:
```suggestion
Both configurations related with standard and non-standard headers can be combined.
```

File: docs/src/main/asciidoc/vertx.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/vertx.adoc:738]:
```suggestion
In this case, the `Forwarded` header will have precedence in presence of both set of headers.
```

File: docs/src/main/asciidoc/vertx.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/vertx.adoc:740]:
```suggestion
Supported forwarding address headers are:
```

File: docs/src/main/asciidoc/vertx.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/vertx.adoc:746]: I'm not sure I'm following the purpose of this list?

Also are the two headers config documented somewhere?

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks!

Considering this has serious security considerations, I will let @stuartwdouglas review it but I did a pass on the doc.

File: docs/src/main/asciidoc/vertx.adoc
Comment (ejba) [reply docs/src/main/asciidoc/vertx.adoc:746]: My intention was to provide a list of headers that Quarkus can use to rewrite request's data. As I didn't find any documentation that expose the list, I decided to include it. 

If you think this list shouldn't be here, I can remove it.

Comment (ejba) [reply docs/src/main/asciidoc/vertx.adoc:746]: The `X-Forwarded-Prefix` header config support is something new. Some ingress proxies (e.g. nginx) generates this header when they remove a prefix in the request's path to keep the original info. However, you have special cases like envoy which it use another header, x-envoy-path if I am not in mistake, for the same purpose. Hence, I let this header "configurable".

Comment (ejba) [reply docs/src/main/asciidoc/vertx.adoc:746]: You can check these config properties [here].(https://github.com/quarkusio/quarkus/blob/f00103df7f39b09c0e46563da33749599b52d1ed/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ProxyConfig.java)

File: docs/src/main/asciidoc/vertx.adoc
Comment (ejba) [reply docs/src/main/asciidoc/vertx.adoc:710]: done

File: docs/src/main/asciidoc/vertx.adoc
Comment (ejba) [reply docs/src/main/asciidoc/vertx.adoc]: done

File: docs/src/main/asciidoc/vertx.adoc
Comment (ejba) [reply docs/src/main/asciidoc/vertx.adoc]: done

File: docs/src/main/asciidoc/vertx.adoc
Comment (ejba) [reply docs/src/main/asciidoc/vertx.adoc]: done

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
@stuartwdouglas doc is good for me on this one. I'll let you validate the security aspect of it.

**Comment by sberyozkin** [comment]
Hi @stuartwdouglas Can you have a quick look please, it LGTM, we can merge if you are happy as well with it

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**
